PetGame/Services/PetService.cs

```diff
@@ -298,31 +325,27 @@ private int CalculateHoursSinceLastActivity(IEnumerable<Activity> PastActivities
             return HoursSinceLastActivity;
         }
 
-        private int CalculateMinutesToNextAction(IEnumerable<Activity> PastActivities, int MinutesToNextAction, int HoursToCheck)
+        private DateTime TimeOfNextAction(IEnumerable<Activity> PastActivities, int MinutesToNextAction, int HoursToCheck)
         {
-            List<Activity> Activities = (List<Activity>)PastActivities;
+            List<Activity> Activities = new List<Activity>(PastActivities);
+            DateTime nextTime = DateTime.Now;
 
             if (Activities.Count == 0)
             {
-                MinutesToNextAction = HoursToCheck * 60;
+                nextTime = DateTime.Now;
             }
             else
             {
-                Activity LastActivity = Activities[0];
-
-                //check the number of minute since the last activity
-                int MinutesSinceLastActivity = 0;
-                TimeSpan span = DateTime.Now.Subtract(LastActivity.Timestamp);
-                MinutesSinceLastActivity = (int)span.TotalMinutes;
+                var LastActivity = Activities.FirstOrDefault(x => x.Type == ActivityType.Feeding || x.Type == ActivityType.Race || x.Type == ActivityType.Training);
 
-                //if it's been more than 5 minutes, the user can perform another action
-                //otherwise, leave the default value in place
-                if (MinutesSinceLastActivity > CooldownLength)
+                if (LastActivity == null)
                 {
-                    MinutesToNextAction = 0;
+                    LastActivity = Activities[0];
```

Comment:
You should not assign anything to `LastActivity`, nor should you use `[0]`, since that's the whole purpose of using `FirstOrDefault`

Reply:
Wait... so if an activity isn't found that's one of the three types that are limited, then you just return the first one? That's not right, in this case, it should allow the user to perform an action.

Reply:
forgot to remove that. it's gone